### ISSUE TITLE
BindingList changes

### ISSUE THREAD

**Issue by pbouzakis** [YuzuJS/WinningJS#22]
We need a helper to easily map binding list changes to ko observable arrays.

Currently this is what we have going on in Beretta:

``` javascript
terms.addEventListener("iteminserted", function (ev) {
    var term = ev.detail.value;
    that.terms.push(term);
});

// TODO Listen for "itemremoved", "itemmutated", "itemchanged"
```

So two issues, need a helper, and need to listen to all events.

**Comment by domenic** [comment]
I think the best approach would be to create a custom Knockout binding that binds to binding lists the same way Knockout currently does to observable arrays.

**Comment by pbouzakis** [comment]
Good idea, will implement something once Roman gets his version of observable arrays into the harness.

**Comment by domenic** [comment]
Meh it looks like we'll need a way to keep two binding lists in sync though, just for viewmodel lists vs. harness lists. Maybe WinRT or WinJS has something already---it's such an obvious thing to do. I'll ask StackOverflow.

**Comment by domenic** [comment]
ORRRR we could let the custom binding take a view model mapper O_o.

``` jade
div(data-win-control="ListView" data-bind="list: theList, listConverter: theConverter")
```

``` js
function TermsListViewModel(termsBindingList) {
    this.list = termsBindingList;
    this.listConverter = function (term) {
        return new TermsViewModel(term);
    };
}
```

I guess maybe this just moves the synchronization code into the custom binding though, which isn't a huge win necessarily. Hmm.

**Comment by domenic** [comment]
ORRR we could do the view model mapping as part of the `itemTemplate` stuff, like I think we are doing currently with `templateRendererFactory` and `componentRendererFactory`.
